frame/assets/src/lib.rs

```diff
@@ -901,6 +786,230 @@ impl<T: Config> Module<T> {
 	}
 }
 
+impl<T: Config> Token<<T::Lookup as StaticLookup>::Source> for Module<T> where
+	T::Balance: MaybeSerializeDeserialize + Debug
+{
+	type Balance = T::Balance;
+	type AssetId = T::AssetId;
+
+	// PUBLIC IMMUTABLES
+
+	/// The total amount of issuance in the system for a specific asset.
+	fn total_issuance(id: Self::AssetId) -> Self::Balance {
+		Self::total_supply(id)
+	}
+
+	fn balance_of(id: Self::AssetId, who: <T::Lookup as StaticLookup>::Source) -> Self::Balance {
+		match T::Lookup::lookup(who) {
+			Ok(acc) => Self::balance(id, acc),
+			Err(_) => Self::Balance::zero(),
+		}
+	}
+
+	/// Returns the remaining number of tokens a spender can spend for a specific asset.
+	#[cfg(feature = "erc20")]
+	fn allowance(id: Self::AssetId, owner: <T::Lookup as StaticLookup>::Source, spender: <T::Lookup as StaticLookup>::Source) -> Self::Balance {
+		let owner = match T::Lookup::lookup(owner) {
+			Ok(acc) => acc,
+			Err(_) => return Self::Balance::zero(),
+		};
+
+		let spender = match T::Lookup::lookup(spender) {
+			Ok(acc) => acc,
+			Err(_) => return Self::Balance::zero(),
+		};
+
+		Approvals::<T>::get(id, (owner, spender))
+	}
+
+	// PUBLIC MUTABLES (DANGEROUS)
+
+	fn transfer(id: Self::AssetId, from: <T::Lookup as StaticLookup>::Source, dest: <T::Lookup as StaticLookup>::Source, amount: Self::Balance) -> DispatchResult {
+		let from = T::Lookup::lookup(from)?;
+		let dest = T::Lookup::lookup(dest)?;
+		let mut from_account = Account::<T>::get(id, &from);
+		from_account.balance = from_account.balance.checked_sub(&amount)
+			.ok_or(Error::<T>::BalanceLow)?;
+
+		if dest == from {
+			return Ok(())
+		}
+
+		Asset::<T>::try_mutate(id, |maybe_details| {
+			let details = maybe_details.as_mut().ok_or(Error::<T>::Unknown)?;
+
+			let mut amount = amount;
+			if from_account.balance < details.min_balance {
+				amount += from_account.balance;
+				from_account.balance = Zero::zero();
+			}
+
+			Account::<T>::try_mutate(id, &dest, |a| -> DispatchResult {
+				let new_balance = a.balance.saturating_add(amount);
+				ensure!(new_balance >= details.min_balance, Error::<T>::BalanceLow);
+				if a.balance.is_zero() {
+					a.is_zombie = Self::new_account(&dest, details)?;
+				}
+				a.balance = new_balance;
+				Ok(())
+			})?;
+
+			match from_account.balance.is_zero() {
+				false => {
+					Self::dezombify(&from, details, &mut from_account.is_zombie);
+					Account::<T>::insert(id, &from, &from_account)
+				}
+				true => {
+					Self::dead_account(&from, details, from_account.is_zombie);
+					Account::<T>::remove(id, &from);
+				}
+			}
+
+			Ok(())
+		})
+	}
+
+	#[cfg(feature = "erc20")]
```

Comment:
I would NOT use feature flags within a runtime pallet. It is a very easy way to create a consensus bug where the native vs wam vs some other version of your runtime are not in sync, and most pipelines would not catch such an issue.

If you want to prevent these functionalities, probably a configuration trait flag would be better with some basic logic which make these functions noop in that case.